modin/pandas/series.py

```diff
@@ -1031,7 +1031,14 @@ def eq(
         Return Equal to of series and `other`, element-wise (binary operator `eq`).
         """
         new_self, new_other = self._prepare_inter_op(other)
-        return super(Series, new_self).eq(new_other, level=level, axis=axis)
+        return new_self._binary_op(
+            "eq",
+            new_other,
+            level=level,
+            fill_value=fill_value,
+            axis=axis,
+            squeeze_other=isinstance(other, (pandas.Series, Series)),
```

Comment:
Why do we need to do `squeeze` in case of a series?

Reply:
I'm using the `Binary.register`/`BinaryDefault.register` operators in the query compiler to implement these functions, which treats the two arguments as `pandas.DataFrame`s when applying the passed function to partitions. I'm not sure if there's a more intelligent way of forcing the arguments to be treated as pandas.Series instead.